DetectorDescription/Core/interface/DDBase.h

```diff
@@ -1,129 +1,131 @@
-  typedef C pimpl_type;
-  typedef DDI::rep_type<N,pimpl_type>* prep_type;
-  typedef std::pair<const N*,bool> def_type;
-  
+
   DDBase() : prep_(nullptr) { }
   virtual ~DDBase() { /*never do this here: if (prep_) delete prep_;*/ }
-      
+  
   const N & name() const { return prep_->name(); }
   
   const N & ddname() const { return prep_->name(); }
   
   std::string toString() const { return prep_->name().fullname(); }
 
-  const typename DDI::rep_traits<N,C>::reference rep() const 
-    { return *(prep_->second); }
+  const typename DDI::rep_traits< N, C >::reference rep() const {
+    return *( prep_->second );
+  }
   
-  typename DDI::rep_traits<N,C>::reference rep() 
-    { return *(prep_->second); }
+  typename DDI::rep_traits< N, C >::reference rep() {
+    return *( prep_->second );
+  }
     
-  const typename DDI::rep_traits<N,C>::reference val() const
-   { if (!isValid()) throw cms::Exception("DDException") << "undefined: " << name(); 
-      return rep();
-    };  
+  const typename DDI::rep_traits< N, C >::reference val() const {
+    if( !isValid()) throw cms::Exception( "DDException" ) << "undefined: " << name(); 
+    return rep();
+  }  
 
-  const typename DDI::rep_traits<N,C>::reference val() 
-   { if (!isValid()) throw cms::Exception("DDException") << "undefined: " << name(); 
-      return rep();
-    };  
+  const typename DDI::rep_traits<N,C>::reference val() {
+    if( !isValid()) throw cms::Exception( "DDException" ) << "undefined: " << name(); 
+    return rep();
+  }
   
-  bool operator==(const DDBase & b) const { return prep_ == b.prep_; }
+  bool operator==( const DDBase & b ) const { return prep_ == b.prep_; }
   // true, if registered or defined
   operator bool() const { return isValid(); }
+  bool operator<( const DDBase & b ) const { return prep_ < b.prep_; }
+  bool operator>( const DDBase & b ) const { return prep_ > b.prep_; }
   
-  bool operator<(const DDBase & b) const { return prep_ < b.prep_; }
-  bool operator>(const DDBase & b) const { return prep_ > b.prep_; }
-  
-  // (name*,true) if defined
-  // (name*,false) if registered but not defined
-  // (0,false) if not there at all
+  // ( name*, true ) if defined
+  // ( name*, false ) if registered but not defined
+  // ( 0, false ) if not there at all
   def_type isDefined() const 
-    {
-      return prep_ ?
-	std::make_pair(&(prep_->name()), bool(prep_->second))
-	:
-	std::make_pair((const N *)nullptr,false);  
-    } 
-   
-  //! true, if the  wrapped pointer is valid
-  bool isValid() const
-    {
-      return prep_ ? bool(prep_->second)
-	: false;
-    } 
+  {
+    return prep_ ?
+      std::make_pair( &( prep_->name()), bool( prep_->second ))
+      :
+      std::make_pair(( const N* )nullptr, false );  
+  }
   
-protected: 
-  prep_type prep_; 
+  //! true, if the wrapped pointer is valid
+  bool isValid() const
+  {
+    return prep_ ? bool( prep_->second )
+      : false;
+  }
+  void create( const N& name, C vals ) {
```

Comment:
Should we consider eventually changing the `create` method to be taking a `std::unique_ptr<C>` since it appears to take ownership of the value?